controllers/machinedeployment_sync.go

```diff
@@ -368,6 +367,24 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 		UnavailableReplicas: unavailableReplicas,
 	}
 
+	if *deployment.Spec.Replicas == status.ReadyReplicas {
+		status.Phase = string(clusterv1.MachineDeploymentPhaseRunning)
+	}
+	if *deployment.Spec.Replicas > status.ReadyReplicas {
+		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingUp)
+	}
+	// same as unavailableReplica, which would have been reset to zero at this point
```

Comment:
```suggestion
	// This is the same as unavailableReplicas, but we have to recalculate because unavailableReplicas would have been reset to zero above if it was negative
```